App/Http/Resolver.php

```diff
@@ -290,18 +224,14 @@ private function downloadVideo($downloadUrl, $saveTo)
         return true;
     }
 
-    /**
-     * @param string $url
-    */
-    private function prepareDownloadLink($url)
+    private function prepareDownloadLink(string $url): array
     {
         $url = $this->getRedirectUrl($url);
-        $url = $this->getRedirectUrl($url);
-        $parts = parse_url($url);
+        $parts = parse_url($url[0]);
```

Comment:
needs attention

Reply:
get redirectUrl now returns an array because getHeader in guzzle 7 returns and array (https://docs.guzzlephp.org/en/stable/psr7.html#accessing-headers)

Reply:
I think the better way would have been to do `return $response->getHeader('Location')[0];` in the get `redirectUrl` method

Reply:
Yeah, I did the same thing